### ISSUE TITLE
Decouple attributes default values from memorized values

### ISSUE THREAD

**Issue by reszelaz** [sardana-org/sardana#1458]
As an introduction to this issue please read #1236 - thanks to @HEnquist! 

The attributes default values should be decoupled from the memorized values. This will have several benefits:

* it will work for non-memorized attributes
* default value changes (controller code changes) will take effect
* it may also facilititate similar solution for the controller properties (not sure)

There are two possibilities of implementation.
1. In the Tango layer, more preciselly, somewhere around `Device_XImpl.init_device()`.
2. In the Sardana core, more preciselly, in the Sardana element constructors.

IMHO, it is better to avoid option 1 just in case one would like to use Sardana without Tango. Then, it won't be necessary to reimplement this logic. So, from now on I will focus on option 2 only.

Regarding the implementation, in the Tango layer we could discover which attributes have memorized values, and pass the list of attributes to the Sardana element constructor. In the constructor then, we could iterate over the attributes and if it does not have the memorized value but it has a default value then write it via the controller. 

Now I can see the following implication: the attributes will be written at the server startup in two groups: first, the attributes never written before with default values, then the memorized attributes previously written at least once.

Following this, I even think it is highly benefitial to avoid restoring the memorized values via Tango, and do both things in the Sardana core and using the Tango DB only as a store of memorized values. Again, it gives a benefit for the ones interested in Sardana without Tango. Another benefit is an easier to control order of restoring the memorized values (see: [the order of reloading memorized attributes is deterministic but complex](https://tango-controls.readthedocs.io/en/latest/development/device-api/ds-guideline/device-server-guidelines.html?highlight=memorized#particular-case-of-a-memorized-attribute) and the problem described in #420).

**Comment by HEnquist** [comment]
I agree with this approach! I very much like the idea of just fetching values from the TangoDB and then let sardana itself write either the default value or the memorized value to the controller. Then it's very clear what will be done, and when it will be done.
